### PR TITLE
Adding Azure.Storage.Common to Package Dependencies and changing DMLib to use package dependencies

### PR DESCRIPTION
**Pull request by amnguye**
Adding Azure.Storage.Common to Package Dependencies and changing DataMovement to use package ref instead of project ref

This PR adds
- Adds Azure.Storage.Common to Package dependencies (added Azure.Storage.Common to the eng/Packages.Data.props)
- Replaces project dependency in the DataMovement packages and uses package dependencies instead

This will allow Data Movement to release more easily outside of the regular storage releases and will only point to the GA releases of the Storage Packages

### CODE REVIEW

**Review by seanmcc-msft — APPROVED** [review]

File: eng/Packages.Data.props [hunk 1]
```diff
@@ -97,6 +97,7 @@
     <PackageReference Update="Azure.Security.KeyVault.Secrets" Version="4.2.0" />
     <PackageReference Update="Azure.Security.KeyVault.Keys" Version="4.2.0" />
     <PackageReference Update="Azure.Security.KeyVault.Certificates" Version="4.2.0" />
+	<PackageReference Update="Azure.Storage.Common" Version="12.15.0" />
```
Comment (schaabs) [inline eng/Packages.Data.props:100]: @weshaggard @tg-msft, should the other storage packages also be updated to use a package reference? Currently I believe this is always pulled in as a project reference.

Comment (weshaggard) [inline eng/Packages.Data.props:100]:
```suggestion
    <PackageReference Update="Azure.Storage.Common" Version="12.15.0" />
```

Comment (seanmcc-msft) [reply eng/Packages.Data.props:100]: @schaabs, the base storage packages (including Common) always ship together.  My understanding is that the release pipeline automatically converts project references to packages references when we release.

The reason we are running into this issue is that we'd like the ability to ship DM Lib separately from the base storage packages.

**Review by weshaggard — APPROVED** [review]
Assuming you plan to independently ship the DataMovement libraires from Common and Blobs then this change is correct.

**Comment by seanmcc-msft** [comment]
> Assuming you plan to independently ship the DataMovement libraires from Common and Blobs then this change is correct.

If we chose to change our workflow and ship DM Lib with the base packages in the future, would we switch this back to a project reference?

**Comment by weshaggard** [comment]
Yes you can change back to project references if you want to ship and have them depend on the exact version you just shipped. If it is OK for them to reference the last version that shipped (i.e. the version in the package.data.props) then you don't have to switch back to package references but you can still ship them together in the same release pipeline.

**Pull request merged**